src/provider/intents.ts

```diff
@@ -0,0 +1,32 @@
+import {Context, ContactContext} from '../client/context';
+import {Intents} from '../client/intents';
+
+/**
+ * TypeScript definitions for FDC3-defined intents.
+ *
+ * These structures are defined by the Intents FDC3 working group. The definitions here are based on current
+ * proposals and are not final. There may be minor differences between the current spec and the definitions here,
+ * in order to support this demo.
+ */
+
+/**
+ * General-purpose intent type
+ */
+export interface Intent<T=string, C extends Context=Context> {
+    type: T;
+    context: C;
+    target?: string;
+}
+
+// Built-in intents
+export type DialCallIntent = Intent<typeof Intents.DIAL_CALL, ContactContext>;
```

Comment:
Phil and I talked. The problem is that, like contexts, we provide a couple of starter ones to get you going (here)[https://fdc3.finos.org/docs/1.0/intents-spec], so we should probably have them in our API

Reply:
IMO, we provide definitions for the "default" contexts, so makes sense to do it for intents too.

But it's true that these types are less useful, since `raiseIntent` (etc) takes each individual field rather than an intent object.

There's no point in defining these within the provider, as the provider has no need to know the shape of specific intents or contexts. They should either be in the client (if we want to make them part of the "official" API), or in the demo app (if we don't).

Reply:
It sounds like you're saying you'll take these out but these are still in